pkg/web/apps/apps_http_handler.go

```diff
@@ -45,28 +47,64 @@ func (a *httpHandler) GetApps(c echo.Context) error {
 	return c.JSON(http.StatusOK, apps)
 }
 
-// GetAppByID returns an app with the ID in JSON format from the AppService
-// TODO: Implement
 //GetAppById returns a app with the ID in JSON format from the AppService
 func (a *httpHandler) GetAppByID(c echo.Context) error {
-	apps, err := a.Service.GetApps()
+	app, err := a.Service.GetAppByID(c.Param("id"))
 
 	if err != nil {
 		return err
 	}
 
-	return c.JSON(http.StatusOK, apps)
+	return c.JSON(http.StatusOK, app)
 }
 
-// UpdateApp returns an app updated with the ID in JSON format from the AppService
-// TODO: Implement
 //UpdateApp returns a app updated with the ID in JSON format from the AppService
-func (a *httpHandler) UpdateApp(c echo.Context) error {
-	apps, err := a.Service.GetApps()
+func (a *httpHandler) UpdateAppVersions(c echo.Context) error {
+	// Validations
+	versions := []models.Version{}
+	errV := json.NewDecoder(c.Request().Body).Decode(&versions)
+
+	// check if the data sent is in the correct format
+	if errV != nil {
+		return httperrors.BadRequest(c, "Invalid data")
+	}
+
+	// Check if versions were sent all the body is empty
+	if len(versions) == 0 {
+		return httperrors.BadRequest(c, "No version(s) was sent.")
+	}
+
+	errUpdate := a.Service.UpdateAppVersions(versions)
+
+	if errUpdate != nil {
+		return httperrors.GetHTTPResponseFromErr(c, errUpdate)
+	}
+
+	return c.JSON(http.StatusOK, "")
+}
+
+//UpdateApp returns a app updated with the ID in JSON format from the AppService
+func (a *httpHandler) UpdateAllAppVersions(c echo.Context) error {
+	versionRes := models.VersionRes{}
+	errV := json.NewDecoder(c.Request().Body).Decode(&versionRes)
+
+	// check if the data sent is in the correct format
+	if errV != nil {
+		return httperrors.BadRequest(c, "Invalid data")
+	}
+
+	app, errApp := a.Service.GetAppByID(c.Param("id"))
```

Comment:
@davidffrench see that we need check if the id informed is a valid one before continuing. 

To do it the most intuitive is to call the method GetAppByID and when it be implemented. It will work. However, I just would like to highlight that since the GetAppByID is returning APP + VERSION, I mean the app object + all versions objects associated with then more queries and unnecessary data will be fetched and bring up to memory. Is it ok for you? Should we care about it now?

Reply:
this logic and the `updateAllAppVersions`should implemented in the service layer as another function which will then be able to just get the app from the repository layer.

Reply:
Regards the layer. Yes, it makes 100% if sense. It is fine. 
However, regards the question is it ok bring all data instead of just the app which is required here? Should we think better in the GetAppByID impl now or can we move forward?

Reply:
if it is in the service layer as a func for example called `updateVersions`, you can use the repository function to just retrieve only the app and not versions